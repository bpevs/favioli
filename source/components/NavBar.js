import React from "react";


export default ({ onSelect }) => <nav><ul>
  <li className="title">ðŸ¤¯</li>
  <li><a
    className="navlink"
    children="Overrides"
    onClick={onSelect}
    href="#overrides" />
  </li>
  <li><a
    className="navlink"
    children="Settings"
    onClick={onSelect}
    href="#settings" />
  </li>
</ul></nav>
