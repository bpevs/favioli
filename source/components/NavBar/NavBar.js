import React from "react";


export const NavBar = ({
  route, onSelect
}) => <nav><ul>
  <li className="title">ðŸ¤¯</li>
  <li><a
    className={"navlink " + (route === "OVERRIDES" ? "active" : "")}
    children="Overrides"
    onClick={onSelect}
    href="#overrides" />
  </li>
  <li><a
    className={"navlink " + (route === "SETTINGS" ? "active" : "")}
    children="Settings"
    onClick={onSelect}
    href="#settings" />
  </li>
</ul></nav>
