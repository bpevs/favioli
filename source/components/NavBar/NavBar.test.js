import { mount, shallow } from "enzyme"
import * as React from "react"
import { NavBar } from "./NavBar"


test("Should Render NavBar", () => {
  const wrapper1 = shallow(<NavBar
    route="OVERRIDES"
    onSelect={jest.fn()}
  />)

  expect(wrapper1).toMatchSnapshot()

  const wrapper2 = shallow(<NavBar
    route="SETTINGS"
    onSelect={jest.fn()}
  />)

  expect(wrapper2).toMatchSnapshot()
});

test("Should run onSelect on click", () => {
  const onSelect = jest.fn()
  const wrapper = mount(<NavBar
    route="SETTINGS"
    onSelect={onSelect}
  />)

  wrapper.find("a").first().simulate("click")
  expect(onSelect).toBeCalled()
});
