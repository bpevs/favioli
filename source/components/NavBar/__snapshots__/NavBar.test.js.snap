// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render NavBar 1`] = `
<nav>
  <ul>
    <li
      className="title"
    >
      ðŸ¤¯
    </li>
    <li>
      <a
        className="navlink active"
        href="#overrides"
        onClick={[MockFunction]}
      >
        Overrides
      </a>
    </li>
    <li>
      <a
        className="navlink "
        href="#settings"
        onClick={[MockFunction]}
      >
        Settings
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Should Render NavBar 2`] = `
<nav>
  <ul>
    <li
      className="title"
    >
      ðŸ¤¯
    </li>
    <li>
      <a
        className="navlink "
        href="#overrides"
        onClick={[MockFunction]}
      >
        Overrides
      </a>
    </li>
    <li>
      <a
        className="navlink active"
        href="#settings"
        onClick={[MockFunction]}
      >
        Settings
      </a>
    </li>
  </ul>
</nav>
`;
